### PR TITLE
Forgot pw functionality

### PR DESCRIPTION
**Pull request by dwulfse**

### CODE REVIEW

File: screens/LoginScreens/ForgottenPasswordScreen.jsx [hunk 1]
```diff
@@ -4,23 +4,40 @@ import FormField from "../../components/Form/FormField.jsx";
...
 import FormButton from "../../components/Form/FormButton.jsx";
 import LoadingWheel from "../../components/Loading/LoadingWheel.jsx";
+import { sendPasswordResetEmail } from "@firebase/auth";
 
 export default function ForgottenPasswordScreen({ navigation }) {
     const [email, setEmail] = useState("");
     const [loading, setLoading] = useState(false);
     const auth = firebase_auth;
 
+    // const handlePasswordReset = async () => {
```
Comment (DanWilkins2107) [inline screens/LoginScreens/ForgottenPasswordScreen.jsx]: Why commented?

File: components/GeneralModal/GeneralModal.jsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { TouchableOpacity, View, ScrollView } from "react-native";
 export default function GeneralModal({ closeModal, children }) {
     return (
         <View className="flex flex-1 justify-center items-center bg-[#FFFFFF90]">
-            <View className={`w-10/12 h-2/5 border-2 rounded-xl  bg-white`}>
+            <View className={`w-10/12 h-fit pb-4 border-2 rounded-xl  bg-white`}>
```
Comment (DanWilkins2107) [inline components/GeneralModal/GeneralModal.jsx:8]: What does fit do?

File: screens/LoginScreens/ForgottenPasswordScreen.jsx [hunk 3]
```diff
@@ -4,23 +4,40 @@ import FormField from "../../components/Form/FormField.jsx";
...
-            alert("Reset Link Sent");
-        } catch (error) {
-            console.log(error);
-            alert(`Password reset failed: ${error.message}`);
-        } finally {
-            setLoading(false);
-        }
+        sendPasswordResetEmail(auth, email)
+            .then(() => {
+                alert("Reset Link Sent");
```
Comment (DanWilkins2107) [inline screens/LoginScreens/ForgottenPasswordScreen.jsx:18]: We need to do visual alerts for these, I'm working on a global alert component atm tho :)

File: screens/LoginScreens/ForgottenPasswordScreen.jsx [hunk 4]
```diff
@@ -4,23 +4,40 @@ import FormField from "../../components/Form/FormField.jsx";
...
+                
+            })
+            .catch((error) => {
+                console.log(error.code, error.message);
```
Comment (DanWilkins2107) [inline screens/LoginScreens/ForgottenPasswordScreen.jsx:22]: Again we need a visual error thing, but our error handling is a bit shit throughout atm

File: screens/LoginScreens/ForgottenPasswordScreen.jsx [hunk 5]
```diff
@@ -4,23 +4,40 @@ import FormField from "../../components/Form/FormField.jsx";
...
+            })
+            .finally(() => {
+                setLoading(false);
```
Comment (DanWilkins2107) [inline screens/LoginScreens/ForgottenPasswordScreen.jsx:25]: We don't want it to go to login if there's an error surely?

**Review by DanWilkins2107 — CHANGES_REQUESTED** [review]
Btw can u text me/ move on Jira once you've done stuff so I can review asap :) I saw on Jira for this one obvs

File: screens/LoginScreens/ForgottenPasswordScreen.jsx (hunk 1 above)
Comment (dwulfse) [reply screens/LoginScreens/ForgottenPasswordScreen.jsx]: cos I'm an idiot. fixed now

File: components/GeneralModal/GeneralModal.jsx (hunk 2 above)
Comment (dwulfse) [reply components/GeneralModal/GeneralModal.jsx:8]: fit the box to the size of the content. then padding at the bottom to make it evenly spaced around the modal so the bottom line isn't basically touching the submit button

File: screens/LoginScreens/ForgottenPasswordScreen.jsx (hunk 4 above)
Comment (dwulfse) [reply screens/LoginScreens/ForgottenPasswordScreen.jsx:22]: quite shocking tbf I agree. pretty patchy too

File: screens/LoginScreens/ForgottenPasswordScreen.jsx (hunk 5 above)
Comment (dwulfse) [reply screens/LoginScreens/ForgottenPasswordScreen.jsx:25]: 100%

File: components/GeneralModal/GeneralModal.jsx (hunk 2 above)
Comment (dwulfse) [reply components/GeneralModal/GeneralModal.jsx:8]: not really relevant to my pr tbf just was annoying me

Comment (DanWilkins2107) [reply components/GeneralModal/GeneralModal.jsx:8]: Works for me :)

**Review by DanWilkins2107 — APPROVED** [review]
Perfect :)

**Pull request merged**